src/ZODB/DB.py

```diff
@@ -645,6 +645,15 @@ def _(c):
         self._mvcc_storage.close()
         del self.storage
         del self._mvcc_storage
+        # clean up references to other DBs
+        self.databases.clear()
+        # clean up the connection pool
+        while self.pool.pop():
+            pass
```

Comment:
Maybe `AbstractConnectionPool` should grow a `clear()` method:  we would then just call it for each one:

```python
        self.pool.clear()
        self.historical_pool.clear()
```